packages/airnode-node/src/api/index.ts

```diff
@@ -0,0 +1,233 @@
+  const ois = config.ois.find((o) => o.title === oisTitle)!;
+  const apiCredentials = config.apiCredentials
+    .filter((c) => c.oisTitle === oisTitle)
+    .map((c) => removeKey(c, 'oisTitle')) as adapter.ApiCredentials[];
+
+  switch (aggregatedApiCall.type) {
+    case 'testing-gateway': {
+      return {
+        endpointName,
+        parameters: sanitizedParameters,
+        ois,
+        apiCredentials,
+        metadata: null,
+      };
+    }
+    case 'regular': {
+      const { airnodeAddress, requesterAddress, sponsorAddress, sponsorWalletAddress, endpointId, id, chainId } =
+        aggregatedApiCall;
+      const chain = config.chains.find((c) => c.id === chainId)!;
+
+      return {
+        endpointName,
+        parameters: sanitizedParameters,
+        ois,
+        apiCredentials,
+        metadata: {
+          airnodeAddress: airnodeAddress,
+          requesterAddress: requesterAddress,
+          sponsorAddress: sponsorAddress,
+          sponsorWalletAddress: sponsorWalletAddress,
+          endpointId: endpointId,
+          requestId: id,
+          chainId: chainId,
+          chainType: chain.type,
+          airnodeRrpAddress: chain.contracts.AirnodeRrp,
+        },
+      };
+    }
+  }
+}
+
+async function signResponseMessage(requestId: string, responseValue: string, config: Config) {
+  const masterHDNode = getMasterHDNode(config);
+  const airnodeWallet = ethers.Wallet.fromMnemonic(masterHDNode.mnemonic!.phrase);
+
+  return await airnodeWallet.signMessage(
+    ethers.utils.arrayify(
+      ethers.utils.keccak256(ethers.utils.solidityPack(['bytes32', 'bytes'], [requestId, responseValue || '0x']))
+    )
+  );
+}
+
+export interface CallApiPayload {
+  readonly config: Config;
+  readonly aggregatedApiCall: AggregatedApiCall;
+}
+
+function verifySponsorWallet(
+  config: Config,
+  aggregatedApiCall: RegularAggregatedApiCall
+): LogsData<ApiCallErrorResponse> | null {
+  const { sponsorAddress, sponsorWalletAddress, id } = aggregatedApiCall;
+  const hdNode = getMasterHDNode(config);
+  if (isValidSponsorWallet(hdNode, sponsorAddress, sponsorWalletAddress)) return null;
+
+  // TODO: Abstract this to a logging utils file
+  const message = `${RequestErrorMessage.SponsorWalletInvalid}, Request ID:${id}`;
+  const log = logger.pend('ERROR', message);
+  return [
+    [log],
+    {
+      success: false,
+      errorMessage: message,
+    },
+  ];
+}
+
+function verifyRequestId(
+  config_: Config,
+  aggregatedApiCall: RegularAggregatedApiCall
+): LogsData<ApiCallErrorResponse> | null {
+  if (isValidRequestId(aggregatedApiCall)) return null;
+
+  const message = `${RequestErrorMessage.RequestIdInvalid}. Request ID:${aggregatedApiCall.id}`;
+  const log = logger.pend('ERROR', message);
+  return [
+    [log],
+    {
+      success: false,
+      errorMessage: message,
+    },
+  ];
+}
+
+export function verifyTemplateId(
+  config_: Config,
```

Comment:
Why the underscore at the end of the variable name? Shouldn't it be at the start for ESLint?
```suggestion
  _config: Config,
```

Reply:
I'm not sure what happened here TBH :sweat_smile: And how that passed a CI check...

Reply:
I think eslint is satisfied with both begin and end underscore.